src/runtime/naming/name.cpp

```diff
@@ -421,20 +422,31 @@ namespace hpx { namespace naming
         ///////////////////////////////////////////////////////////////////////
         std::int64_t replenish_credits(gid_type& gid)
         {
+            typedef std::unique_lock<gid_type::mutex_type> scoped_lock;
+            scoped_lock l(gid);
+            return replenish_credits_locked(l, gid);
+        }
+
+        HPX_EXPORT std::int64_t replenish_credits_locked(
+            std::unique_lock<gid_type::mutex_type>& l, gid_type& gid)
+        {
             std::int64_t added_credit = 0;
 
-            {
-                typedef std::unique_lock<gid_type::mutex_type> scoped_lock;
-                scoped_lock l(gid);
-                HPX_ASSERT(0 == get_credit_from_gid(gid));
+            HPX_ASSERT(0 == get_credit_from_gid(gid));
 
-                added_credit = naming::detail::fill_credit_for_gid(gid);
-                naming::detail::set_credit_split_mask_for_gid(gid);
-            }
+            added_credit = naming::detail::fill_credit_for_gid(gid);
+            naming::detail::set_credit_split_mask_for_gid(gid);
 
             gid_type unlocked_gid = gid;        // strips lock-bit
 
-            return agas::incref(unlocked_gid, added_credit);
+            std::int64_t result = 0;
+            {
+                hpx::util::scoped_unlock<std::unique_lock<gid_type::mutex_type> >
+                    ul(l);
+                 result = agas::incref(unlocked_gid, added_credit);
+            }
+
+            return result;
```

Comment:
Am 10.06.2016 10:19 nachm. schrieb "Hartmut Kaiser" <
notifications@github.com>:
>
> In src/runtime/naming/name.cpp:
>
> >
> >              gid_type unlocked_gid = gid;        // strips lock-bit
> >
> > -            return agas::incref(unlocked_gid, added_credit);
> > +            std::int64_t result = 0;
> > +            {
> > +
hpx::util::scoped_unlock<std::unique_lock<gid_type::mutex_type> >
> > +                    ul(l);
> > +                 result = agas::incref(unlocked_gid, added_credit);
> > +            }
> > +
> > +            return result;
>
> The initial credit is supplied by the underlying heap on gid-creation,
iirc.

Nope. The heap only provides the gid.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.